File: Assets/Luna/MyTextListener.js


pc.MyTextListener = function () {
    

    this.loadingComplete=function()
    {
      FunPlayableSDK.send('loadingComplete');
     console.log("加载完成");
    }

    this.engagement=function()
    {
   FunPlayableSDK.send('engagement');
     console.log("第一次交互");
    }

    this.tutorial=function()
    {
     FunPlayableSDK.send('tutorial');
     console.log("新手引导");
    }
    this.win=function()
    {
      FunPlayableSDK.send('win');
     console.log("胜利");
    }
      this.lose=function()
    {
      FunPlayableSDK.send('lose');
     console.log("失败");
    }
        this.resize=function()
    {
      FunPlayableSDK.send('resize');
     console.log("屏幕旋转");
    }

    this.goStore=function()
    {
     FunPlayableSDK.send('goStore');
     console.log("商店跳转");
    }

    this.finish=function()
    {
       FunPlayableSDK.send('finish');
     console.log("完成试玩打点");
    }
    this.customfunction = function ( message) {
        FunPlayableSDK.send('custom', message);
        console.log("自定义打点");
    }

    this.addnumber = function () {
        return datanumber;
    }
    this.addnumber1 = function () {
        return datanumber1;
    }
    this.addnumber2 = function () {
        return datanumber2;
    }
    this.isluna = function () {
        return lunaOrHtml;
    }
    this.copyAA = function (message) {
        window.$__toCopy = message;
        console.log("copyAA called")
    }

    document.querySelector('#application-canvas').addEventListener("click", function () {
        console.log("canvas clicked")
        setTimeout(() => {
            if (window.$__toCopy) {
                console.log("content being copied")
                var aux = document.createElement("input");
                aux.setAttribute("value", window.$__toCopy);
                document.body.appendChild(aux);
                aux.select();
                document.execCommand("copy");
                document.body.removeChild(aux);
                window.$__toCopy = "";
            }
        }, 100)
    })
}

File: Assets/Luna/MyTextListener.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
// Bridge is available only in luna compilation time, so wrap it in define.
#if UNITY_LUNA
using Bridge;

// It will be replaced in JavaScript with library code.
// Name is important to pass existing class in JS.
[External][Name( "pc.MyTextListener" )]
#endif
public class MyTextListener
{
    public extern string EnterKey(string ctx);
    public extern string LoadingComplete();
    public extern string Engagement();

    public extern string Tutorial();

    public extern string Win();

    public extern string Lose();

    public extern string Resize();

    public extern string Finish();

    public extern string GoStore();

    public extern string Customfunction(string message);

    public extern int Addnumber();
    public extern int Addnumber1();

    public extern int Addnumber2();

    public extern bool Isluna();
    public extern string CopyAA(string message);
}
